tests/test_animation.py

```diff
@@ -99,13 +106,15 @@ def test_send_all_args(self, bot, chat_id, animation_file, animation, thumb_file
 
     @flaky(3, 1)
     @pytest.mark.timeout(10)
+    @pytest.mark.filterwarnings("ignore:.*custom attributes")
```

Comment:
Why is this and the `monkeypatch.delattr`s needed? doesn't monkeypatch work with slottet classes? Having to do that everywhere is bound to lead to problems down the road 😬

Reply:
uh yeah, monkeypatching unfortunately adds them to the `__dict__`, and raises warnings too. And after deleting the attribute, the original method gets added to the `__dict__` of the instance. So the warnings after the end of the test are all from the monkeypatched methods itself.
I only put the delattr thing a few times before realizing its not worth doing that everywhere

Reply:
Wait, does this mean the tests won't work at all, once we drop `__dict__`? o.0 Someone problems of future-us, but still …

Reply:
Yeah I guess so, mostly the Bot tests.

Reply:
Well, that's a bummer … I found [this](https://www.attrs.org/en/stable/glossary.html?highlight=monkeypatch#term-slotted-classes), though - we might leave a note in the tests on this or just directly use such a subclass of `Bot` in the tests with a corresponding note that this currently doesn't have any effect but will be once we drop `__dict__`.

Btw, that link also reminded me of pickle: IIRC you already tested that manually, but it would be good to have a unit test in `TestPicklePersistence` that makes sure that TG objects are picklable (just pickle e.g. a `Message`)

Reply:
Alright will implement all that in the tests